Fix custom renderer

martinRenou opened the pull request:
Custom renderers were broken since my PR "JavaScript -> TypeScript". This was due to the fact that `renderer.js` and `sheet.js` targets were both shipping their own `Handsontable`. Which means that if one defines a custom renderer it will be registered from the `sheet.js` code, but impossible to retrieve from `renderer.js`.

martinRenou commented:
@maartenbreddels would you mind checking this PR? :)

maartenbreddels commented on js/src/extension.ts:
```diff
@@ -2,19 +2,25 @@
+import * as Handsontable from 'handsontable/dist/handsontable.full.js';
+import 'handsontable/dist/handsontable.min.css';
+import '../../src/custom.css';
+
```
This will give an issue with the renderer extension, since you want only 1 handsontable (both extensions the same) we need to make it into a separate module.

martinRenou replied:
I have only one `handsontable`, this one :)

maartenbreddels commented:
I need to read better :)
Anyway, the idea of the separate module was to make sure only 1 handsontable gets loaded, how come there are two?

martinRenou commented:
@maartenbreddels 
So basically, in [this PR](https://github.com/QuantStack/ipysheet/pull/25) I made some changes on the webpack config. After those changes `renderer` and `index` targets were both shipping their own `Handsontable`. And this broke the custom renderers.

So in this PR I fixed it, and `Handsontable` is once again an external dependency for the `index` target and the `renderer` target. So now it works fine.

I'm fine for putting back the `handsontable` target if it was important. I'm not sure why I removed it in the end. I think my issue was that TypeScript was changing the 
```javascript
import * as Handstontable from 'handsontable'
``` 
into 
```javascript
require('../lib/handsontable.js')'
``` 
due to the `handsontable.js` file. And webpack could not resolve the external dependency anymore.

maartenbreddels commented:
I'd rather have it as it was since with this PR Handsontable is always loaded (even if ipysheet is not used).

martinRenou commented:
Ok, I'm not sure it worked due to TypeScript. But I can try

martinRenou commented on js/src/extension.ts:
```diff
@@ -8,8 +8,7 @@ if (window['require'] !== undefined) {
             "*" : {
                 "ipysheet": "nbextensions/ipysheet/index",
-                "handsontable": "nbextensions/ipysheet/handsontable",
-                "handsontable/dist/handsontable.full.js":  "nbextensions/ipysheet/handsontable",
```
Why was this needed?

martinRenou commented on js/src/renderer.ts:
```diff
@@ -1,7 +1,8 @@
 import {extend} from 'lodash';
 import * as pkg from '../package.json';
-import {Handsontable} from './handsontable';
+// @ts-ignore
```
Ignoring with TypeScript (otherwise TypeScript complains that it does not exist)

martinRenou commented:
@maartenbreddels Ok done :) Hope you are happy with the changes. Works like a charm now!

maartenbreddels commented:
Why is the _prefix needed?

martinRenou commented:
There was a conflict between `'./node_module/handsontable'` and `'handsontable'` for TypeScript. But now that I ignore the Handsontable variable in TypeScript it is fine anyway and it does not check the `require`. So I removed the _ prefix
